addons/web/static/src/views/fields/properties/property_definition_selection.js

```diff
@@ -12,18 +12,24 @@ export class PropertyDefinitionSelection extends Component {
         // when we create a new option, it's added in the state
         // when we have finished to edit it (blur / enter) we propagate
         // the new value in the props
-        this.state = useState({ newOption: null });
+        this.state = useState({
+            newOption: null,
+            newOptionIndex: -1,
+        });
```

Comment:
I wouldn't have put it in the state because imo we should not rerender the component if its value changes (for example in onOptionChange, we may update it using event.value, which triggers a render (which will not change anything), before anyways calling props.onOptionsChange which will also trigger a render.

I guess it was just a personal preference of me here, so let's keep it like this.